salt/utils/network.py

```diff
@@ -209,11 +207,15 @@ def get_fqhostname():
         for info in addrinfo:
             # info struct [family, socktype, proto, canonname, sockaddr]
             if len(info) >= 4:
-                l = [info[3]]
+                return info[3]
     except socket.gaierror:
         pass
-
-    return l and l[0] or None
+    except socket.error:
+        pass
```

Comment:
My thinking was: I don't really care what the error is. If I can get the information, great, if not, try something else or give up. Logging makes a ton of sense though.

Reply:
You don't, but normally people do care. Because if everything works, having it at debug level is just fine. But once something weird is happening, and you want it with `salt-support` delivered for examination, and suddenly you have no idea, because there is just `pass`.

Reply:
indeed. it's a solid point. typically I try to catch all errors and be as specific as possible, logging everything I can to debug... for debugging. But it seems to me the style and purpose of grains is try to get things, fire and forget, and then move on if it didn't work.

whatever the case, I did log the errors to debug in the latest push.